### PR TITLE
Add payment request fields for payment requests

### PR DESCRIPTION
**Pull request by subarashii-huch09**
The `barcodeType` and `collectionId` fields have been added to the create payment request endpoint as part of the requirements for implementing digital token redemption. Since we use the public API endpoint `POST /api/payment-requests`, the API documentation needs to be updated to reflect these changes.
[Notion Link ](https://www.notion.so/centrapay/Add-collectionId-barcodeType-fields-to-the-create-payment-request-endpoint-5ce079019cb049778e5de05f52d8bcea?pvs=4)

The API Design doc in Notion is also updated. Here is the [link](https://www.notion.so/centrapay/Create-Payment-Request-7bd0d884c5414823a89cdad9fed60da9?pvs=4#81ffb5f26dc04476a46d8c975c7fe658)

### CODE REVIEW

File: legacy/src/api/payment-requests/payment-requests.md [hunk 1]
```diff
@@ -373,6 +373,8 @@ Payment Activities are created when a Payment Request has been **created**, **pa
 | configId             | String            | The [Merchant Config][] id used to configure the payment options.                                                                                                        |
 | value                | {% dt Monetary %} | The canonical value of the payment request. Must be positive.                                                                                                            |
 | barcode              | String {% opt %}  | The [Scanned Code] to identify the account to attach the payment request to                                                                                              |
+| barcodeType          | String {% opt %}  | An attribute to specify or identify the type of barcode that is being used or processed                                                                                  |
+| collectionId         | String {% opt %}  | The Collection Id                                                                                                                                                        |
```
Comment (seliwe) [inline legacy/src/api/payment-requests/payment-requests.md:377]: Might be more informative if description mentions tokens, e.g. The identifier of the token collection

Comment (subarashii-huch09) [reply legacy/src/api/payment-requests/payment-requests.md:377]: Thanks for the feedback ⭐️  I will update it 👍

File: legacy/src/api/payment-requests/payment-requests.md [hunk 2]
```diff
@@ -373,6 +373,8 @@ Payment Activities are created when a Payment Request has been **created**, **pa
 | configId             | String            | The [Merchant Config][] id used to configure the payment options.                                                                                                        |
 | value                | {% dt Monetary %} | The canonical value of the payment request. Must be positive.                                                                                                            |
 | barcode              | String {% opt %}  | The [Scanned Code] to identify the account to attach the payment request to                                                                                              |
+| barcodeType          | String {% opt %}  | An attribute to specify or identify the type of barcode that is being used or processed                                                                                  |
```
Comment (seliwe) [inline legacy/src/api/payment-requests/payment-requests.md:376]: Suggestion: The owner or origin of the barcode when the barcode is a token

File: legacy/src/api/payment-requests/payment-requests.md
Comment (seliwe) [inline legacy/src/api/payment-requests/payment-requests.md:1]: We should also include the new fields in the Optional Fields table starting line 65

File: legacy/src/api/payment-requests/payment-requests.md [hunk 3]
```diff
@@ -68,6 +68,8 @@ version (documented on this page) and the "legacy" version (documented at
 | -------------------- | ------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
 | patronCodeId         | String             | The id of a [Patron Code][] the payment request is attached to.                                                                                                          |
 | barcode              | String             | [Scanned Code][] used to create the payment request.                                                                                                                     |
+| barcodeType          | String             | The owner or origin of the barcode when the barcode is a token                                                                                                           |
```
Comment (GeekOnGadgets) [inline legacy/src/api/payment-requests/payment-requests.md:71]: I don't think we need to have "when the barcode is token"

**Review by GeekOnGadgets — APPROVED** [review]

File: legacy/src/api/payment-requests/payment-requests.md (hunk 3 above)
Comment (subarashii-huch09) [reply legacy/src/api/payment-requests/payment-requests.md]: Yep 👍, as discussed together, we will change it to  The identifier of barcode type

**Pull request merged**